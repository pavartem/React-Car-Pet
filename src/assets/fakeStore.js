import configureStore from "redux-mock-store";

const fakeStoreObj = {
    cars: [
        {
            id: 1,
            name: 'BMW 320',
            description: 'tedggfdgdgf',
            added: false,
        },
        {
            id: 2,
            name: 'BMW X5',
            description: 'tedggfdgdgf2',
            added: true,
        },
    ],
    loading: false,
    loadingAutoComplete: false,
    autoComplete: [],
};

export const store = configureStore()(fakeStoreObj);


