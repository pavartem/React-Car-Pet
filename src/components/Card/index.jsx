import React from 'react';
import './index.css';

import {useDispatch} from "react-redux";
import {toggleCartItemAction} from "../../store/actions/carActions";

const Card = ({car}) => {

    const dispatch = useDispatch();
    const imgSrc = 'https://upload.wikimedia.org/wikipedia/commons/7/7f/2020_BMW_X7_M_Sport.jpg';
    return (
        <div className='col-md-6'>
            <div className="card">
                <img src={imgSrc} className="card-img-top" alt="test"/>
                <div className="card-body">
                    <h5 className="card-title">{car.name}</h5>
                    <p className="card-text">{car.description}</p>
                    <div
                        onClick={() => dispatch(toggleCartItemAction(car))}
                        className={car.added ? `btn btn-primary disabled` : 'btn btn-primary'}
                    >
                        Add to cart
                    </div>
                </div>
            </div>
        </div>
    );
};

export default Card;
