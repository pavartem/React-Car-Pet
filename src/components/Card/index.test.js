import React from "react";
import {configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Provider} from 'react-redux';
import {store} from '../../assets/fakeStore';
import Card from "./index";


configure({adapter: new Adapter()});

describe('<CardItem/>', () => {

    const car = store.getState().cars[0];
    const wrapper = mount(<Provider store={store}><Card car={car}/></Provider>);

    it('Should render component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('Should render right props for card', () => {
        expect(wrapper.find(<Card car={car}/>)).toBeTruthy();
    });
});
