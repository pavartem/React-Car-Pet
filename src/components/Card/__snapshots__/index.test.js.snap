// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardItem/> Should render component 1`] = `ReactWrapper {}`;
